api/src/main/java/org/apache/iceberg/ExpireSnapshots.java

```diff
@@ -55,6 +55,15 @@
    */
   ExpireSnapshots expireOlderThan(long timestampMillis);
 
+  /**
+   * Retains at least last N snapshots and expires all snapshots older than the given timestamp
```

Comment:
this comment is not in sync with the api. `Expires older than given timestamp` aught to be `Expires all snapshots older than the Nth-last snapshot from current snapshot`

Reply:
I documented the behavior with this, which I think covers the case you're talking about:

> If a snapshot would be expired becuase it is older than the expiration timestamp, but is one of the numSnapshot most recent ancestors of the current state, it will be retained. This will not cause snapshots explicitly identified by id from expiring.